flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecOverAggregate.java

```diff
@@ -231,7 +233,7 @@ private List<OverWindowFrame> createOverWindowFrames(
                     GeneratedAggsHandleFunction genAggsHandler =
                             generator
                                     .needAccumulate()
-                                    .needRetract()
+                                    .needRetract(aggCallNeedRetractions)
```

Comment:
comments : we may not use retract in batch sql, but i found OffsetOverFrame(belongs to BatchExecOverAggregate) use retract function